### ISSUE TITLE
Feature Request: Integrated option to make e-books (epub, mobi, fb2, etc.) from archived pages

### ISSUE THREAD

**Issue by Valporaena** [ArchiveBox/ArchiveBox#752]
## Type 

 - [ ] General question or discussion
 - [x] Propose a brand new feature
 - [ ] Request modification of existing behavior or design

## Describe the ideal specific solution you'd want, and whether it fits into any broader scope of changes

I think it would be cool to have an integrated [ebook-convert](https://manual.calibre-ebook.com/generated/en/ebook-convert.html) tool from Calibre (or something like it - this one is just the best one I know for that purpose) you can access from Archivebox web UI/command line. For a web UI I'm thinking something like: you tick the needed boxes, and where you now have "pull", "reset" and other options you would also have a "convert to e-book" option. Wanting to browse your articles comfortably on a dedicated e-ink/reader device seems like a common thing - would be neat to have an option to convert articles in bulk straight from your archive. And since Archivebox already archives pdf's, this option shouldn't seem out of place.

## How badly do you want this new feature?

 - [ ] It's an urgent deal-breaker, I can't live without it
 - [ ] It's important to add it in the near-mid term future
 - [x] It would be nice to have eventually

---

 - [x] I like ArchiveBox so far / would recommend it to a friend

**Comment by pirate** [comment]
Related: https://github.com/ArchiveBox/ArchiveBox/issues/82

**Comment by Valporaena** [comment]
Yeah, I've tried that: you get and html file, which most e-book reader apps can't properly format. You would need to run that file through Calibre anyway, which is kind of a pain. Or you can make an ebook-convert script, I guess - but I'm completely illiterate when it comes to scripting and programming, unfortunately. :)

I've just recently looked at the Archivebox web UI and thought: wouldn't it be convenient if you could tick all these boxes and just get the files you need from here. Seems like some people thought it would be cool a few years ago too. I get that the project is about archiving first and foremost, but with my particular workflow this kind of thing would be ideal.

**Comment by Valporaena** [comment]
Just want to give a quick update re: the importance of this particular feature.

Turns out at least some modern e-ink readers (Onyx Boox models in particular) have pretty good tools for reading raw html files in ebook form. They need some minor tweaking, but after that no conversion to epub/fb2 is needed, really. I don't know if this is true for Kindle, Kobo or other popular devices though.

**Comment by ralienpp** [comment]
I also need this functionality, though I think that one could consider a slightly different approach. Instead of having ArchiveBox do the conversion, provide an option to define a callback function or some REST endpoint. Once the web page is saved, it will be fed into this external system.

This way the logic is decoupled from that of ArchiveBox and others can come up with their own conversion mechanisms for various purposes. I'd like to turn this into an equivalent of "Send to Kindle";  but I wouldn't want to burden ArchiveBox with the need to understand how the Amazon ecosystem works.

**Comment by pirate** [comment]
Callback webhooks are implemented in v0.8.0 along with the new REST API :)

https://github.com/ArchiveBox/ArchiveBox/pull/1418

Should be enough to build this out externally with N8N/NodeRED/IFTTT/Zapier/etc.

<img src="https://github.com/ArchiveBox/ArchiveBox/assets/511499/fe2ffecc-657d-4eb8-9a16-b3198e859f60" width="400px"/>

**Comment by pirate** [comment]
I also recommend ppl check out https://readeck.org/en/ if you're looking for a solution that offers e-pub export.
